src/priceHistory/PriceHistoryChart.tsx

```diff
@@ -0,0 +1,363 @@
+  position,
+  x,
+  y,
+  value,
+  chartWidth,
+}: {
+  position: 'top' | 'bottom'
+  x: number
+  y: number
+  value: string
+  chartWidth: number
+}) {
+  if (position === 'top') {
+    y = y - 16
+  } else if (position === 'bottom') {
+    y = y + 25
+  }
+  const [adjustedX, setAdjustedX] = useState(x)
+  const horizontalOffset = variables.contentPadding
+  const onLayout = useCallback(
+    ({
+      nativeEvent: {
+        layout: { width },
+      },
+    }: LayoutChangeEvent) => {
+      if (Math.abs(width - chartWidth) > 2) {
+        if (x - width / 2 - horizontalOffset < 0) {
+          setAdjustedX(width / 2 + horizontalOffset)
+        }
+        if (x + width / 2 + horizontalOffset > chartWidth) {
+          setAdjustedX(chartWidth - width / 2 - horizontalOffset)
+        }
+      }
+    },
+    [x]
+  )
+  return (
+    <SvgText
+      /*
+      // @ts-ignore */
+      onLayout={onLayout}
+      fill={colors.gray4}
+      fontSize="14"
+      fontFamily="Inter-Regular"
+      x={adjustedX}
+      y={y}
+      textAnchor="middle"
+    >
+      {value}
+    </SvgText>
+  )
+}
+
+function renderPointOnChart(
+  chartData: Array<{ amount: number | BigNumber; displayValue: string }>,
+  chartWidth: number,
+  color: colors
+) {
+  let lowestRateIdx = 0,
+    highestRateIdx = 0
+  chartData.forEach((rate, idx) => {
+    if (rate.amount > chartData[highestRateIdx].amount) {
+      highestRateIdx = idx
+    }
+    if (rate.amount < chartData[lowestRateIdx].amount) {
+      lowestRateIdx = idx
+    }
+  })
+  return ({ datum, x, y }: { x: number; y: number; datum: { _x: number; _y: number } }) => {
+    const idx = datum._x
+    const result = []
+    switch (idx) {
+      case 0:
+        result.push(
+          <G key={idx + 'dot'}>
+            <Line x1={0} y1={y} x2={chartWidth} y2={y} stroke={colors.gray2} strokeWidth="1" />
+            <Circle cx={x} cy={y} r="4" fill={color} />
+          </G>
+        )
+        break
+
+      case chartData.length - 1:
+        result.push(
+          <G key={idx + 'dot'}>
+            <Circle cx={x} cy={y} r="4" fill={color} />
+          </G>
+        )
+        break
+    }
+    switch (idx) {
+      case highestRateIdx:
+        result.push(
+          <ChartAwareSvgText
+            x={x}
+            y={y}
+            key={idx}
+            value={chartData[highestRateIdx].displayValue}
+            position={'top'}
+            chartWidth={chartWidth}
+          />
+        )
+        break
+
+      case lowestRateIdx:
+        result.push(
+          <ChartAwareSvgText
+            x={x}
+            y={y}
+            key={idx}
+            value={chartData[lowestRateIdx].displayValue}
+            position={'bottom'}
+            chartWidth={chartWidth}
+          />
+        )
+        break
+    }
+
+    switch (result.length) {
+      case 0:
+        return null
+      case 1:
+        return result[0]
+      default:
+        return <>{result}</>
+    }
```

Comment:
can these be condensed into: 
```suggestion
    return (
      <>
        <G key={idx + 'dot'}>
          {idx === 0 && (
            <Line x1={0} y1={y} x2={chartWidth} y2={y} stroke={colors.gray2} strokeWidth="1" />
          )}
          <Circle cx={x} cy={y} r="4" fill={color} />
        </G>
        {[highestRateIdx, lowestRateIdx].includes(idx) && (
          <ChartAwareSvgText
            x={x}
            y={y}
            key={idx}
            value={chartData[highestRateIdx].displayValue}
            position={idx === highestRateIdx ? 'top' : 'bottom'}
            chartWidth={chartWidth}
          />
        )}
      </>
    )
```

Reply:
Went with a slightly modified version in f3829090717396e4bf8865f8fb223976f2362d75, that appears to be working well.

Reply:
Actually looking at the snapshots I see empty  `<g />` elements added with this method.